inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+sim1 <- netsim(est, param, init, control)
+
+## tergmLite = TRUE: use reduced cross-sectional edgelist
+control <- control.net(type = "SI", nsteps = 100, nsims = 1, tergmLite = TRUE)
+sim2 <- netsim(est, param, init, control)
+```
+
+## Network Update
+
+Prior to EpiModel 2.0, the base functions of EpiModel were built using functions both from ```EpiModel``` and ```network``` packages. This intermingling of functions made it difficult for the end user to tease out where network functionality ended and EpiModel began, hampering understanding of the EpiModel workflow and the ability to build module extensions. In order to make the distinction between the two packages explicit and to make the purely EpiModel workflow clearer, a new base function, ```nwupdate.net``` has been introduced. This function pulls all interactions with the network object from base functions into ```nwupdate.net``` and is run at the end of the simulation. By segmenting the EpiModel workflow in this way, base EpiModel modules are now easier to navigate and make clear the output needed for module extensions.
+
+## Helper Functions
+
+Several helper functions have been added to EpiModel that make it clear the ```$``` indexing operations that are being done on the master ```dat``` object during simulation. These functions belong to the ```get``` and ```set``` family of functions and act on the various lists contained within the master ```dat``` object, including but not limited to the attribute, epi, control and param lists. These functions take as input, at mininum, the master ```dat``` object and the item (as a character string) to "get" or "set" ("age" or "group" in the attribute list or "inf.prob" in the parameter list for exampe).
+
+### Set Functions
+
+The ```set``` group of functions sets the value of an item on a sublist of the ```dat``` object; these functions are often used at the end of a modules sequence to update attributes or for tracking of epidemic statistics. As an example: 
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from one-Group infection module:
+dat <- set_epi(dat, "si.flow", at, nInf)
+```
+
+Here, we want to record the number of new infected - ```nInf``` -  at the current time step ```at```. We do so using the ```set_epi``` helper function and record the flow of succeptible to infected on to the master ```dat``` object. All set functions follow this naming scheme, with "set_" followed by the sublist of interest. Note: the set functions returns an updated ```dat``` object.
+
+### Get Functions
+
+The ```get``` group of functions retrieve the value of an item on a sublist of the ```dat``` objects, and are most often used when retrieving input parameters of ```init```, ```param``` and ```control``` objects or for querying current attribute or epidemic tracker lists. As an example:
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from One-Group arrivals module
+a.rate <- get_param(dat, "a.rate")
+nOld <- get_epi(dat, "num", index)
```

Comment:
same here for the EpiModel 1.8 code (if we decide to add it)

`a.rate <- dat[["attr"]]["a.rate"]]`